WIP: Upgrade jQuery 1.7.2 to 2.2.x

dan-f opened the pull request:
[FEDX-25](https://openedx.atlassian.net/browse/FEDX-25)

dan-f commented:
@andy-armstrong feel free to pile on now :)

andy-armstrong commented on common/static/common/js/common_libraries.js:
```diff
@@ -1 +1 @@
-define(['domReady!', 'jquery', 'backbone', 'underscore', 'gettext']);
+define(['domReady!', 'jquery', 'jquery-migrate', 'backbone', 'underscore', 'gettext']);
```
@dan-f I think this is a mistake, as 'jquery-migrate' is actually mapped to 'jquery' itself, right? I'm seeing errors in Studio trying to load jquery-migrate:

![image](https://cloud.githubusercontent.com/assets/5985072/13230911/5fb2d168-d975-11e5-9d66-95d5dafa02d7.png)

dan-f replied:
You're right. I tried to revert this change in a fixup commit but it must have not been applied. I'll fix it.

andy-armstrong commented:
jenkins run all

dan-f commented:
jenkins run all

andy-armstrong commented:
@symbolist FYI

cahrens commented on cms/static/js/views/modals/course_outline_modals.js:
```diff
@@ -533,13 +534,14 @@ define(['jquery', 'backbone', 'underscore', 'gettext', 'js/views/baseview',
         afterRender: function () {
             AbstractEditor.prototype.afterRender.call(this);
-            this.setValue(this.model.get('format'));
+            this.setValue(this.model.get('format') || 'notgraded');
```
@dan-f did you make this change? Can you explain it?

cahrens commented on cms/static/js/spec/views/pages/course_outline_spec.js:
```diff
@@ -243,6 +244,16 @@ define(["jquery", "common/js/spec_helpers/ajax_helpers", "common/js/components/u
                 ]);
 
+                // Create a mock Course object as the JS now expects it.
+                window.course = new Course({
```
I should clean this up in afterEach.

andy-armstrong commented on lms/djangoapps/courseware/features/problems.feature:
```diff
@@ -223,10 +223,9 @@ Feature: LMS.Answer problems
         When I answer a "<ProblemType>" problem "<InitialCorrectness>ly"
         Then my "<ProblemType>" answer is marked "<InitialCorrectness>"
-        And I input an answer on a "<ProblemType>" problem "<OtherCorrectness>ly"
+        And I reset the problem
```
Reviewers: it took me a long time to be sure of this, but I believe that the Lettuce test was written incorrectly as it didn't match the comment. Before it wasn't testing the reset behavior because it did no testing after it performed the reset. Instead if was verifying that the answer was not marked when providing an incorrect answer, which isn't true.

An interesting question is why this test used to pass before this upgrade. I'll investigate that tomorrow. Either way, I believe that the test as now written is verifying what the comment (and test name) suggests that it is.

dan-f replied:
Nice find.

> An interesting question is why this test used to pass before this upgrade.

This should be the official slogan of our jQuery upgrade.

dan-f commented:
@andy-armstrong @cahrens FYI - I fixed all the box_choy tests early on, but it looks like things are failing pretty seriously currently. I'm wondering if this is the result of a bad rebase or refactor. I'm going to spend my energy tomorrow getting bok_choy to work again... if I fixed it initially, I should be able to get it working again.

andy-armstrong commented:
jenkins run quality

andy-armstrong commented:
@dan-f Did you mean to make this PR against the feature branch?

andy-armstrong commented:
jenkins run all

dan-f commented:
@andy-armstrong the feature branch was created after this PR. I'll re-create it.

andy-armstrong commented:
jenkins run all

dan-f commented on cms/static/js/views/course_info_update.js:
```diff
@@ -80,6 +80,7 @@ define(["js/views/baseview", "codemirror", "js/models/course_update",
         },
 
         onSave: function(event) {
+            debugger
```
Delete this :)

dan-f commented on cms/static/js/views/course_info_update.js:
```diff
@@ -111,6 +112,7 @@ define(["js/views/baseview", "codemirror", "js/models/course_update",
         },
 
         onCancel: function(event) {
+            debugger
```
Delete this :)

dan-f commented:
I'd like to verify that my bok_choy fix actually fixes things, then I'd like to do a rebase and re-open this PR against our feature branch.

dan-f commented:
Recreated at https://github.com/edx/edx-platform/pull/11665
